clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/UpsertGoogleHttpLoadBalancerAtomicOperation.groovy

```diff
@@ -163,6 +163,9 @@ class UpsertGoogleHttpLoadBalancerAtomicOperation extends UpsertGoogleLoadBalanc
           break
       }
       targetProxyExists = existingProxy as Boolean
+      if (existingProxy && GCEUtil.getLocalName(existingProxy.getUrlMap()) != description.urlMapName) {
+        throw new IllegalStateException("Listener with name ${existingRule.getName()} already exists and points to a different url map than ${description.urlMapName}")
```

Comment:
It might be helpful to specify the name of the url map that the listener is pointing to .